### PR TITLE
Allow users to configure subgraph urls and subtopics

### PR DESCRIPTION
**Pull request by axiomatic-aardvark**
### Description
This PR introduces changes that are needed to support [integration tests for the POI Radio](https://github.com/graphops/poi-radio/pull/2) (and integration testing for Radios in general)

Changes:
- Switch to using env vars for `REGISTRY_SUBGRAPH` and `NETWORK_SUBGRAPH`
- Allow Radio developers to override the default subtopics by passing another argument to `GossipAgent::new()`
- Removes POI Radio from `examples/` folder (removes the whole folder actually, but it's being added back in the [ping-pong Radio PR](https://github.com/graphops/graphcast-sdk/pull/33))
### Issue link (if applicable)
Will solve https://github.com/graphops/graphcast-sdk/issues/32

### CODE REVIEW

File: src/gossip_agent/message_typing.rs [hunk 1]
```diff
@@ -53,11 +54,13 @@ fn prepare_nonces(
...
 
 pub async fn get_indexer_stake(address: String) -> Result<BigUint, anyhow::Error> {
-    Ok(
-        query_network_subgraph(NETWORK_SUBGRAPH.to_string(), address.clone())
-            .await?
-            .indexer_stake(),
-    )
+    dotenv().ok();
+
+    let network_subgraph = env::var("NETWORK_SUBGRAPH").expect("No NETWORK_SUBGRAPH set.");
```
Comment (hopeyen) [inline src/gossip_agent/message_typing.rs:59]: maybe let's process the environmental variables upon radio startup and initialize gossip agent with them? This way setup warnings is shown earlier on, and this particular function doesn't need to access env everytime

File: src/gossip_agent/message_typing.rs [hunk 2]
```diff
@@ -183,12 +186,14 @@ impl GraphcastMessage {
...
     pub async fn valid_sender(&self) -> Result<&GraphcastMessage, anyhow::Error> {
-        let indexer_address = query_registry_indexer(
-            REGISTRY_SUBGRAPH.to_string(),
-            self.recover_sender_address()?,
-        )
-        .await?;
-        if query_network_subgraph(NETWORK_SUBGRAPH.to_string(), indexer_address.clone())
+        dotenv().ok();
+
+        let registry_subgraph = env::var("REGISTRY_SUBGRAPH").expect("No REGISTRY_SUBGRAPH set.");
```
Comment (hopeyen) [inline src/gossip_agent/message_typing.rs:191]: same here

File: src/gossip_agent/message_typing.rs (hunk 1 above)
Comment (axiomatic-aardvark) [reply src/gossip_agent/message_typing.rs:59]: that's a good idea, or maybe if none are provided we can set these as default ones on the SDK side

Comment (axiomatic-aardvark) [reply src/gossip_agent/message_typing.rs:59]: Sure, I can add them as fields on `GossipAgent`, although it feels like there are a lot of setup fields already, maybe post-MVP we can use a yaml config file for this (we can give the users the option to construct with fields using `new` or load a yaml config with `from_file` or something), I'll add an Issue for that

Comment (axiomatic-aardvark) [reply src/gossip_agent/message_typing.rs:59]: I like it because it's good to keep env vars outside the SDK

Comment (axiomatic-aardvark) [reply src/gossip_agent/message_typing.rs]: Done ✅

**Review by hopeyen — DISMISSED** [review]
💯

**Review by hopeyen — APPROVED** [review]

**Review by hopeyen — APPROVED** [review]

**Pull request merged**